### PR TITLE
escaping quotes, added requirements and gitignore

### PR DESCRIPTION
**Pull request by couillonnade**
- Added a requirement.txt file to install required module
- Added a .gitignore file to ignore phi3v folder
- Fixed escaping issue un line 700 and 730

For line 738, escaping was not easy so I introduced intermediate variables.

Error references:

```
File "Phi-3-Vision-MLX/main.py", line 700
list_str = [f".. code-block:: python\n :caption: {filename}\n\n{textwrap.indent(code, " ")}" for filename, code in zip(list_doc, list_str)]
^^^
SyntaxError: f-string: unmatched '('
```

and 

```
SyntaxError: f-string: unmatched '('
File "Phi-3-Vision-MLX/main.py", line 738
print(f'### Prompt ###\n{"\n".join(map(str.strip, prompt)).strip()}\n### Images ###\n{'\n'.join(map(str, images)) if images else "None"}\n### Output ###') if verbose else None
^
SyntaxError: unexpected character after line continuation character
```

### CODE REVIEW

**Comment by JosefAlbers** [comment]
@couillonnade,

First of all, thank you so much for your pull request! I am super excited to see the first contribution to my repository, and I appreciate your taking the time to make the changes. I apologize however there was a bit of a mix-up on my end, and I accidentally made a commit that overwrites the changes in your pull request before seeing your pull request. 

I would really like to incorporate your changes into the repo though. Would you be able to rebase your branch on top of the latest version of the main branch and resubmit your pull request? If you need any assistance with the rebase process or have any questions, please don't hesitate to ask! 

Thank you again for your contribution!

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Automatically generated by https://github.com/damnever/pigar.
+
+datasets==2.20.0
+huggingface-hub==0.23.4
+matplotlib==3.9.0
+mlx==0.15.1
```
Comment (JosefAlbers) [inline requirements.txt]: The model's behavior's been a bit weird with mlx 0.15.1 (released just a few days ago). I cannot yet pinpoint the exact cause, so for now, I am fixing the mlx version to 0.15.0 in this repository. Additionally, I am keeping the rest of the requirements.txt as is for now.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+phi3v/
```
Comment (JosefAlbers) [inline .gitignore]: Agree! Also:
```
phi3v/
quantized_phi3v/
adapters/
*.egg-info
*.json
.DS_STORE
.DS_Store
__pycache__/
```

File: main.py [hunk 3]
```diff
@@ -697,7 +697,7 @@ def to_lora(layer):
 def _load_text(root_url, list_doc, is_code=False, join=True):
     list_str = [response.text for file_url in list_doc if (response := requests.get(f'{root_url}/{file_url}')).status_code == 200]
     if is_code:
-        list_str = [f".. code-block:: python\n    :caption: {filename}\n\n{textwrap.indent(code, "    ")}" for filename, code in zip(list_doc, list_str)]
+        list_str = [f".. code-block:: python\n    :caption: {filename}\n\n{textwrap.indent(code, '    ')}" for filename, code in zip(list_doc, list_str)]
```
Comment (JosefAlbers) [inline main.py]: Absolutely! However, I had replaced _load_text() with a simpler load_text() function that simply returns the text content of a file or URL. I apologize once again for not catching your pull request before making my commit.

File: main.py [hunk 4]
```diff
@@ -735,7 +735,11 @@ def _apply_chat_template(prompt, images, verbose):
         img_prompt = ''
     prompt = [prompt] if isinstance(prompt, str) else prompt
     prompt = [f"<|user|>\n{img_prompt}{i}<|end|>\n<|assistant|>\n" for i in prompt]
-    print(f'### Prompt ###\n{"\n".join(map(str.strip, prompt)).strip()}\n### Images ###\n{'\n'.join(map(str, images)) if images else "None"}\n### Output ###') if verbose else None
+    if verbose:
+        prompt_str = "\n".join(map(str.strip, prompt)).strip()
+        images_str = "\n".join(map(str, images)) if images else "None"
+        print(f'### Prompt ###\n{prompt_str}\n### Images ###\n{images_str}\n### Output ###')
+
```
Comment (JosefAlbers) [inline main.py]: Agree 100%!

File: main.py (hunk 3 above)
Comment (couillonnade) [reply main.py]: No worries, the new version is clearly better!

**Comment by couillonnade** [comment]
Closed and new PR #2 submitted.

**Pull request closed without merging**